### ISSUE TITLE
Transitioner NullReferenceException (sometimes) when SelectedIndex changes

### ISSUE THREAD

**Issue by Talha-T** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#937]
```csharp
// Get clicked item
var navigationItem = (NavigationItem) (sender as FrameworkElement)?.DataContext;
var viewModel = (MainWindowViewModel) DataContext;

Debug.Assert(navigationItem != null, "navigationItem != null");
    
// Set the placeholder (first item) to content
// Pages is the ItemsSource of Transitioner
viewModel.Pages[0] = navigationItem.Content;
// Navigate to 0th index
transitiner.SelectedIndex = 0; // Exception being thrown here
```
Exception Details:
```
System.NullReferenceException oluştu
  HResult=0x80004003
  İleti=Nesne başvurusu bir nesnenin örneğine ayarlanmadı.
  Kaynak=<Özel durum kaynağı değerlendirilemiyor>
  StackTrace:
   konum MaterialDesignThemes.Wpf.Transitions.Transitioner.ActivateFrame(Int32 selectedIndex, Int32 unselectedIndex)
   konum MaterialDesignThemes.Wpf.Transitions.Transitioner.OnSelectionChanged(SelectionChangedEventArgs e)
   konum System.Windows.Controls.Primitives.Selector.InvokeSelectionChanged(List`1 unselectedInfos, List`1 selectedInfos)
   konum System.Windows.Controls.Primitives.Selector.SelectionChanger.End()
   konum System.Windows.Controls.Primitives.Selector.SelectionChanger.SelectJustThisItem(ItemInfo info, Boolean assumeInItemsCollection)
   konum System.Windows.Controls.Primitives.Selector.OnSelectedIndexChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
   konum System.Windows.DependencyObject.OnPropertyChanged(DependencyPropertyChangedEventArgs e)
   konum System.Windows.FrameworkElement.OnPropertyChanged(DependencyPropertyChangedEventArgs e)
   konum System.Windows.DependencyObject.NotifyPropertyChange(DependencyPropertyChangedEventArgs args)
   konum System.Windows.DependencyObject.UpdateEffectiveValue(EntryIndex entryIndex, DependencyProperty dp, PropertyMetadata metadata, EffectiveValueEntry oldEntry, EffectiveValueEntry& newEntry, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType)
   konum System.Windows.DependencyObject.InvalidateProperty(DependencyProperty dp, Boolean preserveCurrentValue)
   konum System.Windows.Data.BindingExpressionBase.Invalidate(Boolean isASubPropertyChange)
   konum System.Windows.Data.BindingExpression.TransferValue(Object newValue, Boolean isASubPropertyChange)
   konum System.Windows.Data.BindingExpression.ScheduleTransfer(Boolean isASubPropertyChange)
   konum MS.Internal.Data.ClrBindingWorker.NewValueAvailable(Boolean dependencySourcesChanged, Boolean initialValue, Boolean isASubPropertyChange)
   konum MS.Internal.Data.PropertyPathWorker.UpdateSourceValueState(Int32 k, ICollectionView collectionView, Object newValue, Boolean isASubPropertyChange)
   konum MS.Internal.Data.ClrBindingWorker.OnSourcePropertyChanged(Object o, String propName)
   konum MS.Internal.Data.PropertyPathWorker.OnPropertyChanged(Object sender, PropertyChangedEventArgs e)
   konum System.Windows.WeakEventManager.ListenerList`1.DeliverEvent(Object sender, EventArgs e, Type managerType)
   konum System.ComponentModel.PropertyChangedEventManager.OnPropertyChanged(Object sender, PropertyChangedEventArgs args)
   konum LearnCSharp.ViewModels.MainWindowViewModel.OnPropertyChanged(String propertyName) C:\Users\seher\RiderProjects\LearnCSharp\LearnCSharp\ViewModels\MainWindowViewModel.cs içinde: 49. satır
   konum LearnCSharp.ViewModels.MainWindowViewModel.set_Index(Int32 value) C:\Users\seher\RiderProjects\LearnCSharp\LearnCSharp\ViewModels\MainWindowViewModel.cs içinde: 41. satır
   konum LearnCSharp.MainWindow.NavigationItem_OnMouseDown(Object sender, MouseButtonEventArgs e) C:\Users\seher\RiderProjects\LearnCSharp\LearnCSharp\MainWindow.xaml.cs içinde: 76. satır
   konum System.Windows.Input.MouseButtonEventArgs.InvokeEventHandler(Delegate genericHandler, Object genericTarget)
   konum System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   konum System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   konum System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   konum System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   konum System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
   konum System.Windows.UIElement.RaiseEvent(RoutedEventArgs args, Boolean trusted)
   konum System.Windows.Input.InputManager.ProcessStagingArea()
   konum System.Windows.Input.InputManager.ProcessInput(InputEventArgs input)
   konum System.Windows.Input.InputProviderSite.ReportInput(InputReport inputReport)
   konum System.Windows.Interop.HwndMouseInputProvider.ReportInput(IntPtr hwnd, InputMode mode, Int32 timestamp, RawMouseActions actions, Int32 x, Int32 y, Int32 wheel)
   konum System.Windows.Interop.HwndMouseInputProvider.FilterMessage(IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   konum System.Windows.Interop.HwndSource.InputFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   konum MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   konum MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   konum System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   konum System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   konum System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   konum MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   konum MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   konum System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   konum System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   konum System.Windows.Application.RunDispatcher(Object ignore)
   konum System.Windows.Application.RunInternal(Window window)
   konum System.Windows.Application.Run(Window window)
   konum System.Windows.Application.Run()
   konum LearnCSharp.App.Main()
```

**Comment by Talha-T** [comment]
[Ref]( https://github.com/ButchersBoy/MaterialDesignInXamlToolkit/blob/2740f14a814896d42032ae0013b765a8a0ec04c3/MaterialDesignThemes.Wpf/Transitions/Transitioner.cs#L194)

`slide` value seems to be null.
And it is because `GetSlide()` returns null ([ref](https://github.com/ButchersBoy/MaterialDesignInXamlToolkit/blob/2740f14a814896d42032ae0013b765a8a0ec04c3/MaterialDesignThemes.Wpf/Transitions/Transitioner.cs#L166))

**Comment by Keboo** [comment]
Fixed by #1242
